### PR TITLE
Fix interrupt reply

### PR DESCRIPTION
**Pull request by garlandz-db**
issue: https://github.com/ipython/ipykernel/issues/1100#issuecomment-1448133194 

Changes in PR
* Added reply content properties for interrupt_request
* `send_interupt_children` => `send_interrupt_children` renaming
* Added test coverage to verify statuses are received

### CODE REVIEW

File: ipykernel/kernelbase.py [hunk 1]
```diff
@@ -892,20 +892,25 @@ def _send_interupt_children(self):
...
-                    os.killpg(pgid, SIGINT)
-                    return
-                except OSError:
-                    pass
-            try:
+                os.killpg(pgid, SIGINT)
+            else:
                 os.kill(pid, SIGINT)
-            except OSError:
-                pass
```
Comment (garlandz-db) [inline ipykernel/kernelbase.py:903]: move out OSError try except

File: ipykernel/tests/test_ipkernel_direct.py [hunk 2]
```diff
@@ -90,8 +90,21 @@ async def test_comm_info_request(ipkernel):
 
 
 async def test_direct_interrupt_request(ipkernel):
-    reply = await ipkernel.test_shell_message("interrupt_request", {})
+    reply = await ipkernel.test_control_message("interrupt_request", {})
```
Comment (garlandz-db) [inline ipykernel/tests/test_ipkernel_direct.py:93]: shell => control

**Comment by garlandz-db** [comment]
@blink1073 how do i add the label? i dont see an option

File: ipykernel/kernelbase.py (hunk 1 above)
Comment (blink1073) [reply ipykernel/kernelbase.py:903]: This is a change in behavior.  I think we want to do something like:

```python
try:
    os.killpg(pgid, SIGINT)
except OSError:
    os.kill(pid, SIGINT)
    raise
os.kill(pid, SIGINT)
````

Comment (garlandz-db) [reply ipykernel/kernelbase.py:903]: oops youre right

**Comment by blink1073** [comment]
@ccordoba12 this fix breaks one of the `spyder-kernels` tests, but I believe it is the correct behavior.

**Comment by ccordoba12** [comment]
@impact27, could you take a look at this one, please?

**Comment by impact27** [comment]
In spyder kernel we want to be able to raise an interrupt from a thread on the main thread. This is done by 
`self.kernel._send_interupt_children()`

I guess we just need to rename it `self.kernel._send_interrupt_children()` but that means we will have to change the minimum ipykernel version so this works

**Comment by blink1073** [comment]
We could make this backward compatible by having `_send_interupt_children` warn and call `_send_interrupt_children`.

**Comment by ccordoba12** [comment]
Well, the call to `_send_interupt_children()` is only present in our master branch, which hasn't been released yet. So, I think it's simpler if we skip the failing test so that this PR passes. Then we switch to the new method name and increase our minimal IPykernel supported, as @impact27 suggested.

**Comment by ccordoba12** [comment]
@blink1073, our test suite should pass now thanks to PR https://github.com/spyder-ide/spyder-kernels/pull/446. I went for a simpler solution: detect if the kernel has `_send_interupt_children` and if not use `_send_interrupt_children`.

**Comment by blink1073** [comment]
Confirmed, thanks all!  I'll make a release early next week.

**Pull request merged**